File: prefect_square/perfect_square.cpp
#include <bits/stdc++.h>
using namespace std;

bool isSqrt(long double n)
{
    if(n<=0)
        return false;
    if(ceil(sqrt(n)) == floor(sqrt(n)) )
        return true;
    return false;
}

int main() {
    
    ios_base::sync_with_stdio(false);
    long double n = 1000000;
    
    
    if(isSqrt(n))
        cout << fixed << setprecision(2) << n << " Is perfect square\n";
    else
        cout << fixed << setprecision(2) << n << " Is not perfect square\n";
    
    
    return 0;
}

File: prefect_square/perfect_square.py
from math import sqrt

def isSquare(n):
	return sqrt(n) % 1 == 0

if __name__ == "__main__":
	n = int(input())
	if isSquare(n):
		print("This is a perfect square")
	else:
		print("This is not a perfect square")
